packages/roosterjs-editor-plugins/lib/ContentEdit/features/markdownFeatures.ts

```diff
@@ -0,0 +1,127 @@
+import { PluginKeyboardEvent } from 'roosterjs-editor-types';
+import {
+    Browser,
+    toggleBold,
+    toggleItalic,
+    toggleUnderline,
+    toggleStrikethrough,
+} from 'roosterjs/lib';
+import { Editor, ContentEditFeature, Keys } from 'roosterjs-editor-core';
+
+const findTriggerCharactersAndCheckBetween = (triggerChar: string) => (
+    event: PluginKeyboardEvent,
+    editor: Editor
+) => {
+    let selObj = window.getSelection();
+    let anchorNode = selObj.anchorNode;
+    let genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(' ' + triggerChar);
+    let alterationRequired = false;
+    let stringAdjust = 2;
+
+    if (
```

Comment:
compute this value once, and check it each time. goes for both of these conditionals